tests/xdc/nexus_state_replication_test.go

```diff
@@ -494,6 +495,216 @@ func (s *NexusStateReplicationSuite) TestNexusCallbackReplicated() {
 	}
 }
 
+func (s *NexusStateReplicationSuite) TestNexusOperationBufferedCompletionReplicated() {
+	ctx := testcore.NewContext()
+	ns := s.createGlobalNamespace()
+	taskQueue := "tq"
+
+	allowCompletion := atomic.Bool{}
+
+	h := nexustest.Handler{
+		OnStartOperation: func(
+			ctx context.Context,
+			service, operation string,
+			input *nexus.LazyValue,
+			options nexus.StartOperationOptions,
+		) (nexus.HandlerStartOperationResult[any], error) {
+			if !allowCompletion.Load() {
+				return &nexus.HandlerStartOperationResultAsync{
+					OperationID: "test",
+				}, nil
```

Comment:
You should return an internal handler error here instead, otherwise we won't retry this request and the completion would need to be delivered async via the callback. The code that you have under this condition is essentially unreachable.

Reply:
Alternatively, you could use the callback path to issue the completion but I'm not convinced that's simpler. Feel free to try that though.

Reply:
Switched to internal handler error